SpringProject/src/main/java/com/gradproject2019/conferences/service/ConferenceServiceImpl.java

```diff
@@ -75,4 +85,12 @@ private void checkConferenceExists(Long conferenceId) {
             throw new ConferenceNotFoundException();
         }
     }
+
```

Comment:
(sorry if double comment) 

I suggest having this check at the web/resource layer. 

Having it here means that an authentication token needs to be provided to do these business functions. If another part of the application wants to edit/save/delete a conference using these APIs they will need to have to provide a token, despite them being invoked presumably after authentication

The Business layer shouldn't really handle `authentication` related things, it should only be aware of business related functions - it really helps to keep each 'tier' of the app separate

Reply:
So I know that the business layer translates to the Service layer since it does the business logic, but what layer does the web/resource layer translate to? from the "web" part of the name I'm guessing it's the front end (which would then require each action that requires authentication to hit to both an authorisation endpoint and the endpoint that carries out said action). But if it's the back end then the authentication would have to happen in the Controller layer which we've been told to keep logic out of?

Reply:
In this case it refers to your `*Controllers` - its the layer that takes HTTP requests and does something with them. It is the API (_interface_) that your application has to the wider internet (FE, or other 3rd party applications). You _cannot_ rely on a front end to provide security

Keeping logic out the `Controller` is an important consideration - if its what you've been told by your helpers you should follow their advice. 

However when you think about it, your controllers already have logic in - 
`@RequestBody UserPatchRequestDto userPatchRequestDto, UserRequestDto userRequestDto, @PathVariable("id") Long userId`

This Deserializes a HTTP request into java objects (and does type checking) - you don't pass a stream of bytes say into the business layer. I would suggest creating an `abstract class` called `BaseController` or something that can hande authentication across _all_ controllers with a common `checkAuthentication(UUID token)` method